### DISCUSSION TITLE
Alpine crashes when setting el.value to a file input

### DISCUSSION THREAD

**Discussion by nerdoc** [alpinejs/alpine#4374] [7. Bugs (archived)] [answered]
I am using https://tetraframework.com, which internally uses Alpine.js for creating components. I'm not completely sure if the bug I am experiencing is a Tetra bug or an Alpine one, but as I can't find the bug in Tetra, it seems to be more generic.

Tetra creates Alpine.js components on the fly, and uses `x-model` data bindings, e.g. here (I tried to snip away everything that is not needed):
```html
<div 
    tetra-component="common__default__profile_master_data_editor" 
    x-bind="__rootBind"
     ... 
    x-data="common__default__profile_master_data_editor('{...snip...}')"
>
      <form enctype="multipart/form-data">
        <input type="file" name="avatar" class="form-control" accept="image/*" x-model="avatar" id="id_avatar">
            <button @click="submit()">Save</button>
      </form>
    </div>
```
When loading the page with this component (there are many other inputs), I get an error in the console:
```
Uncaught DOMException: An attempt was made to use an object that is not, or is no longer, usable
    bindInputValue http://localhost:8000/static/tetra/js/alpinejs.cdn.js:1357
    ....
```
The error occurs in the [end of the function bindInputValue()](https://github.com/alpinejs/alpine/blob/ab743bc5668f018db5d410cad4e48b755696ad9e/packages/alpinejs/src/utils/bind.js#L78):

https://github.com/alpinejs/alpine/blob/ab743bc5668f018db5d410cad4e48b755696ad9e/packages/alpinejs/src/utils/bind.js#L78

The direct assignment of any value to `el.value` is forbidden (due to security reasons).

So I think that, for file inputs, Alpine has to create another if/else branch (analogue to selects, checkbox, boolean etc).

I am not capable of doing this, as my Js knowledge is only limited.
Maybe Tetra has to produce some code that prevents creating a `x-model`attr in a file type input - but I wanted to ask here if anyone sees this too as a bug in Alpine.

I can make the error message disappear by adding an empty clause:
```javascript
    ...
    } else if (el.type === "file") {
        // do nothing.
    } else {
    ...
```
But I don't know all the impacts this will produce.

**Comment by ekwoka** [comment, marked as answer]
You can't use x-model on a file input, because their semantics are just totally different from many other input types.

You can handle the change event to get the files into your component, and bind files to the input if needed.

**Reply by nerdoc** [reply]
Ok, that's a technical misunderstanding on my side then, thanks.
